src/App.jsx

```diff
@@ -1,7 +1,75 @@
+import { useState } from 'react';
+import RestaurantList from './RestaurantList';
+
+import TextBox from './TextBox';
+
+function updateRestaurantInfo(state, name, value) {
+  return {
+    ...state,
+    restaurantInfo: {
+      [name]: value,
+    },
+  };
+}
+
+function addRestaurant(state) {
+  const { restaurants, restaurantInfo } = state;
+  return {
+    restaurants: [
+      ...restaurants,
+      restaurantInfo,
+    ],
+    restaurantInfo: {
+      name: '',
+      category: '',
+      address: '',
+    },
+  };
+}
+
 export default function App() {
+  const initialState = {
+    restaurantInfo: {
+      name: '',
+      address: '',
+      category: '',
+    },
+    restaurants: [{
+      id: 1,
+      name: '마녀주방',
+      address: '한식',
+      category: '서울시 강남구',
+    }, {
+      id: 2,
+      name: '시카고피자',
+      address: '양식',
+      category: '이태원동',
+    }, {
+      id: 3,
+      name: '키와미',
+      address: '일식',
+      category: '분당구 정자동',
+    }],
+  };
+
+  const [state, setState] = useState(initialState);
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setState((previousState) => (updateRestaurantInfo(previousState, name, value)));
```

Comment:
변수의 수명이 긴 이름일수록 더 구체적인 이름을 가져야 합니다. 왜냐하면 변수를 선언한 곳과 거리가 멀기 때문에 이름만 봐도 바로 알 수 있어야 하기 때문입니다. 반대로 변수의 수명이 짧을수록 더 간단한 이름을 사용해야 합니다. 변수의 수명이 짧은 이름은 아주 잠깐만 사용할 것이기 때문에 오래 기억할 필요가 없고 변수를 선언한 곳과 가깝기 때문에 쉽게 알 수 있기 때문입니다.

```suggestion
    setState((s) => updateRestaurantInfo(s, name, value));
```

Reply:
> 변수의 수명이 긴 이름일수록 더 구체적인 이름을 가져야 합니다. 왜냐하면 변수를 선언한 곳과 거리가 멀기 때문에 이름만 봐도 바로 알 수 있어야 하기 때문입니다. 반대로 변수의 수명이 짧을수록 더 간단한 이름을 사용해야 합니다. 변수의 수명이 짧은 이름은 아주 잠깐만 사용할 것이기 때문에 오래 기억할 필요가 없고 변수를 선언한 곳과 가깝기 때문에 쉽게 알 수 있기 때문입니다.

수명이 짧을 수록 간단한 이름을 짓는 생각은 못해봤습니다! 감사합니다.